src/Backups/BackupImpl.cpp

```diff
@@ -838,6 +841,7 @@ void BackupImpl::writeFile(const String & file_name, BackupEntryPtr entry)
         if (use_archives)
         {
             LOG_TRACE(log, "Adding file {} to archive", adjusted_path);
+            std::lock_guard lock{mutex};
```

Comment:
We still will copy data with this lock to archive. Is it intended?

Reply:
It's easier to implement it this way here, and writing to an archive is processed strictly in one thread anyway (see the function `supportsWritingInMultipleThreads`)